src/core/Akka.Remote/Transport/Streaming/NetworkStreamTransport.cs

```diff
@@ -293,12 +292,8 @@ protected virtual Task<AssociationHandle> CreateInboundAssociation(Stream stream
             Socket socket = CreateSocket(endpoint);
             Settings.ConfigureSocket(socket);
 
-            var connectTask = Task.Factory.FromAsync(socket.BeginConnect, socket.EndConnect, remoteAddress.Host, port, null);
-
-            var completedTask = await Task.WhenAny(connectTask, Task.Delay(Settings.ConnectionTimeout));
```

Comment:
awaiting Task.WhenAny did not throw an exception when connectTask failed. So we were creating the association even in case of failure and it was being disassociated shortly after.